pageserver/src/rocksdb_storage.rs

```diff
@@ -165,9 +203,27 @@ impl RocksObjectStore {
         let mut opts = Self::get_rocksdb_opts();
         opts.create_if_missing(true);
         opts.set_error_if_exists(true);
-        let db = rocksdb::DB::open(&opts, &path)?;
+        let obj_store = Self::new(conf, opts)?;
+        Ok(obj_store)
+    }
 
-        let obj_store = RocksObjectStore { _conf: conf, db };
+    fn new(conf: &'static PageServerConf, mut opts: rocksdb::Options) -> Result<RocksObjectStore> {
+        let path = conf.workdir.join("rocksdb-storage");
+        let gc = Arc::new(GarbageCollector::new());
+        let gc_ref = gc.clone();
+        opts.set_compaction_filter("ttl", move |_level: u32, _key: &[u8], _val: &[u8]| {
```

Comment:
should be `key`, not `_key`, since it is used.

Reply:
Fixed